hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieMetadataBootstrap.java

```diff
@@ -261,12 +260,8 @@ private void bootstrapAndVerifyFailure() throws Exception {
     writeConfig = getWriteConfig(true, true);
     initWriteConfigAndMetatableWriter(writeConfig, true);
     syncTableMetadata(writeConfig);
-    try {
-      validateMetadata(testTable);
-      Assertions.fail("Should have failed");
-    } catch (IllegalStateException e) {
-      // expected
-    }
+    Assertions.assertThrows(AssertionFailedError.class, () -> validateMetadata(testTable),
```

Comment:
Should there be an assert or validation in the non-test code itself? I think asserting `AssertionFailedError` is like swallowing any exception. We should assert some specific exception.

Reply:
Yeah, then we need to throw a specific expection from the `validateMetadata`.

Reply:
I just revert it back to what it was.